contracts/adapters/mellow/MellowClaimerAdapter.sol

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Gearbox Protocol. Generalized leverage for DeFi protocols
+// (c) Gearbox Foundation, 2024.
+pragma solidity ^0.8.23;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC4626} from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import {AbstractAdapter} from "../AbstractAdapter.sol";
+import {IMellowClaimer} from "../../integrations/mellow/IMellowClaimer.sol";
+import {
+    IMellowMultiVault,
+    IEigenLayerWithdrawalQueue,
+    Subvault,
+    MellowProtocol
+} from "../../integrations/mellow/IMellowMultiVault.sol";
+import {IMellowClaimerAdapter, MellowMultivaultStatus} from "../../interfaces/mellow/IMellowClaimerAdapter.sol";
+import {MellowWithdrawalPhantomToken} from "../../helpers/mellow/MellowWithdrawalPhantomToken.sol";
+
+import {NotImplementedException} from "@gearbox-protocol/core-v3/contracts/interfaces/IExceptions.sol";
+
+/// @title Mellow Claimer adapter
+/// @notice Implements logic allowing CAs to interact with a Mellow Claimer, which allows them to claim mature withdrawals.
+contract MellowClaimerAdapter is AbstractAdapter, IMellowClaimerAdapter {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    bytes32 public constant override contractType = "ADAPTER::MELLOW_CLAIMER";
+    uint256 public constant override version = 3_10;
+
+    /// @notice Maps a staked phantom token to the multiVault it represents
+    mapping(address stakedPhantomToken => address multiVault) public phantomTokenToMultivault;
+
+    /// @notice Set of allowed multiVaults
+    EnumerableSet.AddressSet private _allowedMultivaults;
+
+    uint256 public constant MAX_ASSETS_BUFFER = 100;
+
+    constructor(address _creditManager, address _claimer) AbstractAdapter(_creditManager, _claimer) {}
+
+    /// @notice Accepts transferred pending assets in the Multivault for the credit account
+    /// @dev During a MultiVault rebalance, a withdrawal request may end up with some withdrawal indices
+    ///      in a transferred state. In order to correctly reflect the in pendingAssetsOf(), these
+    ///      transfers need to be accepted, so this call must always be coupled with a withdrawal request.
+    function multiAccept(address multiVault, uint256[] calldata subvaultIndices, uint256[][] calldata indices)
+        external
+        creditFacadeOnly
+        returns (bool)
+    {
+        if (!_allowedMultivaults.contains(multiVault)) revert MultivaultNotAllowedException();
+        _execute(
+            abi.encodeCall(IMellowClaimer.multiAcceptAndClaim, (multiVault, subvaultIndices, indices, address(0), 0))
+        );
```

Comment:
Seems, could be replaced with  `_claim(multiVault, subVaultIndices, indices, maxAssets);`
Also suggest to move `if (!_allowedMultiVaults.contains(multiVault)) revert MultivaultNotAllowedException();`check into function `_claim`